File: examples/Fortran/openmp_converted/Get_started_vector_addition_solution_omp.f90
program vector_addition
    use iso_fortran_env, only : INT32, REAL64
    use openacc
use omp_lib
    implicit none

    integer(kind=INT32 ), parameter              :: system_size = 1e5
    real   (kind=REAL64), dimension(system_size) :: s, c, array_sum
    real   (kind=REAL64)                         :: fortran_pi
    integer(kind=INT32 )                         :: i

    fortran_pi = acos(-1.0_real64)

    !$acc data create(s(:), c(:)) copyout(array_sum(:))
!$omp target data map(from:array_sum(:)) map(alloc:s(:),c(:))

    !$acc parallel
!$omp target teams
    !$acc loop
!$omp loop
    do i = 1, system_size
        s(i) = sin(i*fortran_pi/system_size) * sin(i*fortran_pi/system_size)
        c(i) = cos(i*fortran_pi/system_size) * cos(i*fortran_pi/system_size)
    enddo
    !$acc end parallel
!$omp end target teams

    !$acc parallel
!$omp target teams
    !$acc loop
!$omp loop
    do i = 1, system_size - 1 
        array_sum(i) = s(i) + c(system_size - i)
    enddo
    !$acc end parallel
!$omp end target teams

    !$acc end data
!$omp end target data
    
    write(0,"(a10,f10.8)") "sum(42) = ",array_sum(42)
end program vector_addition 

! Code was translated using: intel-application-migration-tool-for-openacc-to-openmp Get_started_vector_addition_solution.f90

File: examples/C/openmp_converted/Get_started_vector_addition_solution_omp.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
int main(void)
{
    int size=1e5;
    double s[size];
    double c[size];
    double sum[size];
    #pragma acc data create(s, c) copyout(sum)
#pragma omp target data map(from:sum) map(alloc:s,c)
    {
    #pragma acc parallel
#pragma omp target teams
    {
        #pragma acc loop
#pragma omp loop
        for (int i=0; i<size;++i)
        {
            s[i] = sin(i*M_PI/size)*sin(i*M_PI/size);
            c[i] = cos(i*M_PI/size)*cos(i*M_PI/size);
        }    
    }
    #pragma acc parallel
#pragma omp target teams
    {
        #pragma acc loop
#pragma omp loop
        for (int i=1; i<size ; ++i)
            sum[i] = s[i] + c[size - i]; 
    }
    }// end of structured data region
    printf("sum[42] = %f\n", sum[42]);
}


// Code was translated using: /home/very/bin/acc2mp -async=nowait -specify-language=C -generate-report Get_started_vector_addition_solution.c
